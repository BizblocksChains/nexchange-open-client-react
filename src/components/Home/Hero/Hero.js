import React, { Component } from 'react';
import { I18n, Trans } from 'react-i18next';

import ExchangeWidget from './ExchangeWidget/ExchangeWidget';
import CoinPrices from './CoinPrices/CoinPrices';
import ErrorAlert from './ErrorAlert/ErrorAlert';

import styles from './Hero.scss';

class Hero extends Component {
  render() {
    return (
      <I18n ns="translations">
        {t => (
          <div className={styles.hero}>
            <ErrorAlert />

            <div className="container">
              <div className="row">
                <div className="col-xs-12">
                  <div className={styles.brand}>
                    <h1>{t('hero.1')}</h1>
                    <Trans i18nKey="hero.2">
                      <h2>
                        Simple. <span className="text-green">Secure</span>. Transparent.
                      </h2>
                    </Trans>
                  </div>
                </div>

                <ExchangeWidget store={this.props.store} />
              </div>
            </div>

            <CoinPrices />
          </div>
        )}
      </I18n>
    );
  }
}

export default Hero;
