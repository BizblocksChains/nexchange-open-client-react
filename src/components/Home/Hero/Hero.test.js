import React from 'react';
import { shallow } from 'enzyme';
import Hero from './Hero';

describe('Hero', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<Hero />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
