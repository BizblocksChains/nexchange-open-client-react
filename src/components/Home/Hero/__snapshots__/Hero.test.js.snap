// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(ErrorAlert) />,
        <div
          className="container"
>
          <div
                    className="row"
          >
                    <div
                              className="col-xs-12"
                    >
                              <div
                                        className="brand"
                              >
                                        <h1>
                                                  hero.1
                                        </h1>
                                        <Trans
                                                  i18nKey="hero.2"
                                        >
                                                  <h2>
                                                            Simple. 
                                                            <span
                                                                      className="text-green"
                                                            >
                                                                      Secure
                                                            </span>
                                                            . Transparent.
                                                  </h2>
                                        </Trans>
                              </div>
                    </div>
                    <Connect(ExchangeWidget)
                              store={undefined}
                    />
          </div>
</div>,
        <Connect(CoinPrices) />,
      ],
      "className": "hero",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-xs-12"
            >
                        <div
                                    className="brand"
                        >
                                    <h1>
                                                hero.1
                                    </h1>
                                    <Trans
                                                i18nKey="hero.2"
                                    >
                                                <h2>
                                                            Simple. 
                                                            <span
                                                                        className="text-green"
                                                            >
                                                                        Secure
                                                            </span>
                                                            . Transparent.
                                                </h2>
                                    </Trans>
                        </div>
            </div>
            <Connect(ExchangeWidget)
                        store={undefined}
            />
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs-12"
>
                <div
                                className="brand"
                >
                                <h1>
                                                hero.1
                                </h1>
                                <Trans
                                                i18nKey="hero.2"
                                >
                                                <h2>
                                                                Simple. 
                                                                <span
                                                                                className="text-green"
                                                                >
                                                                                Secure
                                                                </span>
                                                                . Transparent.
                                                </h2>
                                </Trans>
                </div>
</div>,
              <Connect(ExchangeWidget)
                store={undefined}
/>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="brand"
>
                  <h1>
                                    hero.1
                  </h1>
                  <Trans
                                    i18nKey="hero.2"
                  >
                                    <h2>
                                                      Simple. 
                                                      <span
                                                                        className="text-green"
                                                      >
                                                                        Secure
                                                      </span>
                                                      . Transparent.
                                    </h2>
                  </Trans>
</div>,
                "className": "col-xs-12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1>
                      hero.1
</h1>,
                    <Trans
                      i18nKey="hero.2"
>
                      <h2>
                                            Simple. 
                                            <span
                                                                  className="text-green"
                                            >
                                                                  Secure
                                            </span>
                                            . Transparent.
                      </h2>
</Trans>,
                  ],
                  "className": "brand",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "hero.1",
                    },
                    "ref": null,
                    "rendered": "hero.1",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <h2>
                        Simple. 
                        <span
                                                className="text-green"
                        >
                                                Secure
                        </span>
                        . Transparent.
</h2>,
                      "i18nKey": "hero.2",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Simple. ",
                          <span
                            className="text-green"
>
                            Secure
</span>,
                          ". Transparent.",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Simple. ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Secure",
                            "className": "text-green",
                          },
                          "ref": null,
                          "rendered": "Secure",
                          "type": "span",
                        },
                        ". Transparent.",
                      ],
                      "type": "h2",
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "store": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ErrorAlert) />,
          <div
            className="container"
>
            <div
                        className="row"
            >
                        <div
                                    className="col-xs-12"
                        >
                                    <div
                                                className="brand"
                                    >
                                                <h1>
                                                            hero.1
                                                </h1>
                                                <Trans
                                                            i18nKey="hero.2"
                                                >
                                                            <h2>
                                                                        Simple. 
                                                                        <span
                                                                                    className="text-green"
                                                                        >
                                                                                    Secure
                                                                        </span>
                                                                        . Transparent.
                                                            </h2>
                                                </Trans>
                                    </div>
                        </div>
                        <Connect(ExchangeWidget)
                                    store={undefined}
                        />
            </div>
</div>,
          <Connect(CoinPrices) />,
        ],
        "className": "hero",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="col-xs-12"
              >
                            <div
                                          className="brand"
                            >
                                          <h1>
                                                        hero.1
                                          </h1>
                                          <Trans
                                                        i18nKey="hero.2"
                                          >
                                                        <h2>
                                                                      Simple. 
                                                                      <span
                                                                                    className="text-green"
                                                                      >
                                                                                    Secure
                                                                      </span>
                                                                      . Transparent.
                                                        </h2>
                                          </Trans>
                            </div>
              </div>
              <Connect(ExchangeWidget)
                            store={undefined}
              />
</div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-xs-12"
>
                  <div
                                    className="brand"
                  >
                                    <h1>
                                                      hero.1
                                    </h1>
                                    <Trans
                                                      i18nKey="hero.2"
                                    >
                                                      <h2>
                                                                        Simple. 
                                                                        <span
                                                                                          className="text-green"
                                                                        >
                                                                                          Secure
                                                                        </span>
                                                                        . Transparent.
                                                      </h2>
                                    </Trans>
                  </div>
</div>,
                <Connect(ExchangeWidget)
                  store={undefined}
/>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="brand"
>
                    <h1>
                                        hero.1
                    </h1>
                    <Trans
                                        i18nKey="hero.2"
                    >
                                        <h2>
                                                            Simple. 
                                                            <span
                                                                                className="text-green"
                                                            >
                                                                                Secure
                                                            </span>
                                                            . Transparent.
                                        </h2>
                    </Trans>
</div>,
                  "className": "col-xs-12",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h1>
                        hero.1
</h1>,
                      <Trans
                        i18nKey="hero.2"
>
                        <h2>
                                                Simple. 
                                                <span
                                                                        className="text-green"
                                                >
                                                                        Secure
                                                </span>
                                                . Transparent.
                        </h2>
</Trans>,
                    ],
                    "className": "brand",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "hero.1",
                      },
                      "ref": null,
                      "rendered": "hero.1",
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": <h2>
                          Simple. 
                          <span
                                                    className="text-green"
                          >
                                                    Secure
                          </span>
                          . Transparent.
</h2>,
                        "i18nKey": "hero.2",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Simple. ",
                            <span
                              className="text-green"
>
                              Secure
</span>,
                            ". Transparent.",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "Simple. ",
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Secure",
                              "className": "text-green",
                            },
                            "ref": null,
                            "rendered": "Secure",
                            "type": "span",
                          },
                          ". Transparent.",
                        ],
                        "type": "h2",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "store": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
